styx-common/src/main/java/com/spotify/styx/api/Middlewares.java

```diff
@@ -151,17 +159,27 @@ private static GoogleIdToken verifyIdToken(String s) {
     };
   }
 
+  @AutoMatter
   public interface AuthContext {
+    boolean hasAuthHeader();
+    boolean isValid();
```

Comment:
When auth is enforced, this doesn't seem that useful. I was thinking we could just throw an exception and handle that somewhere else… but maybe that's harder.